lib/nixd/src/Server.cpp

```diff
@@ -44,6 +56,20 @@ void Server::onDocumentDidOpen(
   const std::string &Contents = Params.textDocument.text;
 
   addDocument(File, Contents, encodeVersion(Params.textDocument.version));
+  lspserver::elog("nix settings: {0}", nix::settings.storeUri.get());
+  auto NixStore = nix::openStore();
+  auto State = std::make_unique<nix::EvalState>(nix::Strings(), NixStore);
+  try {
+    State->parseExprFromFile(File.str());
+  } catch (const nix::ParseError &PE) {
+    lspserver::elog("got nix parse error! {0}", PE.msg());
```

Comment:
Ideally, we can catch nix exceptions & retrieve backtrace in this exception (provided by nix itself).